### PR TITLE
Feature/refactor gecko

### PR DESCRIPTION
**Pull request by Tlaster**

### CODE REVIEW

File: buildSrc/src/main/kotlin/Versions.kt [hunk 1]
```diff
@@ -43,6 +43,8 @@ object Versions {
     const val zxing = "4.3.0"
     const val maskWalletCore = "0.4.0"
     const val jodaTime = "2.10.13"
+    const val gecko = "97.+"
```
Comment (qdsfdhvh) [inline buildSrc/src/main/kotlin/Versions.kt]: remove it?

File: extension/src/androidMain/kotlin/com/dimension/maskbook/extension/repository/ExtensionRepository.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    init {
+        controller.installExtensions(
+            id = "info@dimension.com",
+            url = "resource://android/assets/web_extensions/Maskbook/",
+        )
+        controller.onNavigate = {
+            onNavigate(it)
+        }
+        scope.launch {
+            // launch {
```
Comment (qdsfdhvh) [inline extension/src/androidMain/kotlin/com/dimension/maskbook/extension/repository/ExtensionRepository.kt]: remove note code

File: extension/src/androidMain/kotlin/com/dimension/maskbook/extension/utils/MessageChannel.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                    )
+                )
+            )
+            return channel.receive()
+        } finally {
+            channel.close()
+        }
+    }
+
+    // suspend fun executeMessage(
```
Comment (qdsfdhvh) [inline extension/src/androidMain/kotlin/com/dimension/maskbook/extension/utils/MessageChannel.kt]: remove note code

File: wallet/src/androidMain/res/values/styles.xml [hunk 4]
```diff
@@ -1,17 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources xmlns:tools="http://schemas.android.com/tools">
 
-    <style name="ComposeActivityTheme" parent="android:Theme.DeviceDefault.NoActionBar">
```
Comment (qdsfdhvh) [inline wallet/src/androidMain/res/values/styles.xml:4]: remove this styles.xml?

**Review by qdsfdhvh — APPROVED** [review]
LGTM

File: common/gecko/src/androidMain/kotlin/com/dimension/maskbook/common/gecko/WebContentController.kt [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+                hasUserGesture: Boolean,
+                isSameDomain: Boolean,
+                isRedirect: Boolean,
+                isDirectNavigation: Boolean,
+                isSubframeRequest: Boolean
+            ): RequestInterceptor.InterceptionResponse? {
+                if (isSubframeRequest) {
+                    return null
+                }
+                return if (onNavigate(uri)) {
```
Comment (JoeSteven) [inline common/gecko/src/androidMain/kotlin/com/dimension/maskbook/common/gecko/WebContentController.kt:81]: might cause bugs with release build , reference [#111](https://github.com/DimensionDev/Mask-Android/pull/111)

**Pull request merged**